### ISSUE TITLE
bug: Nested stack paramater error when referencing the output from a peer stack

### ISSUE THREAD

**Issue by redterror** [localstack/localstack#4308]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Localstack appears to incorrectly look up outputs in nested stacks when referenced by their peers / siblings.  Cloudformation snippit:

`parent.yml`:
```
Resources:
  VPCStack:
    Type: AWS::CloudFormation::Stack
    Properties:
      TemplateURL: !Sub ${TemplatePath}/vpc.yaml

  AuroraStack:
    Type: AWS::CloudFormation::Stack
    Properties:
      TemplateURL: !Sub ${TemplatePath}/aurora.yaml
      Parameters:
        VPC: !GetAtt VPCStack.Outputs.VPC
        Subnets: !GetAtt VPCStack.Outputs.PublicSubnets
        AutoPause: !Ref AuroraAutoPause
        MinCapacity: !Ref AuroraMinCapacity
        MaxCapacity: !Ref AuroraMaxCapacity
        DatabaseName: !Ref AuroraDatabaseName
        Username: !Ref AuroraUsername
        Port: !Ref AuroraPort
        SnapshotName: !Ref AuroraSnapshotName
        BackupRetentionPeriod: !Ref AuroraBackupRetentionPeriod
```

`vpc.yml`, output block:

```
Outputs:
  VPC:
    Description: A reference to the created VPC
    Value: !Ref VPC

  PublicSubnets:
    Description: A reference to the public subnet in the 1st & 2nd Availability Zones
    Value: !Join [",", [!Ref PublicSubnet1, !Ref PublicSubnet2]]

  PrivateSubnets:
    Description: A reference to the Private subnet in the 1st & 2nd Availability Zones
    Value: !Join [",", [!Ref PrivateSubnet1, !Ref PrivateSubnet2]]

  PublicSubnet1:
    Description: A reference to the public subnet in the 1st Availability Zones
    Value: !Ref PublicSubnet1

  PublicSubnet2:
    Description: A reference to the public subnet in the 2nd Availability Zones
    Value: !Ref PublicSubnet2

  PrivateSubnet1:
    Description: A reference to the private subnet in the 1st Availability Zones
    Value: !Ref PrivateSubnet1

  PrivateSubnet2:
    Description: A reference to the private subnet in the 2nd Availability Zones
    Value: !Ref PrivateSubnet2
```

It appears that localstack internally passes the entire stringified dict / hash output object into the `SubnetIds` parameter of the AuroraStack. 

Localstack debug log snippit:
```
localstack_1  | 2021-07-15T18:35:43:WARNING:localstack.utils.cloudformation.template_deployer: Error calling <bound method ClientCreator._create_api_method.<locals>._api_call of <botocor
e.client.RDS object at 0x7ff6c2026050>> with params: {'DBSubnetGroupDescription': 'ecclesia-localstack-None-1f0b5606', 'SubnetIds': "[{'OutputKey': 'VPC', 'OutputValue': 'vpc-8c9fc0f0',
'Description': 'A reference to the created VPC', 'ExportName': ''}, {'OutputKey': 'PublicSubnets', 'OutputValue': 'subnet-af223282,subnet-fc1f4a72', 'Description': 'A reference to the pu
blic subnet in the 1st & 2nd Availability Zones', 'ExportName': ''}, {'OutputKey': 'PrivateSubnets', 'OutputValue': 'subnet-03d13c44,subnet-2444c28b', 'Description': 'A reference to the
Private subnet in the 1st & 2nd Availability Zones', 'ExportName': ''}, {'OutputKey': 'PublicSubnet1', 'OutputValue': 'subnet-af223282', 'Description': 'A reference to the public subnet
in the 1st Availability Zones', 'ExportName': ''}, {'OutputKey': 'PublicSubnet2', 'OutputValue': 'subnet-fc1f4a72', 'Description': 'A reference to the public subnet in the 2nd Availabili
ty Zones', 'ExportName': ''}, {'OutputKey': 'PrivateSubnet1', 'OutputValue': 'subnet-03d13c44', 'Description': 'A reference to the private subnet in the 1st Availability Zones', 'ExportN
ame': ''}, {'OutputKey': 'PrivateSubnet2', 'OutputValue': 'subnet-2444c28b', 'Description': 'A reference to the private subnet in the 2nd Availability Zones', 'ExportName': ''}]", 'DBSub
netGroupName': 'sng-5783ca4b'} for resource: {'Type': 'AWS::RDS::DBSubnetGroup', 'Properties': {'DBSubnetGroupDescription': 'ecclesia-localstack-None-1f0b5606', 'SubnetIds': "[{'OutputKe
y': 'VPC', 'OutputValue': 'vpc-8c9fc0f0', 'Description': 'A reference to the created VPC', 'ExportName': ''}, {'OutputKey': 'PublicSubnets', 'OutputValue': 'subnet-af223282,subnet-fc1f4a
72', 'Description': 'A reference to the public subnet in the 1st & 2nd Availability Zones', 'ExportName': ''}, {'OutputKey': 'PrivateSubnets', 'OutputValue': 'subnet-03d13c44,subnet-2444
c28b', 'Description': 'A reference to the Private subnet in the 1st & 2nd Availability Zones', 'ExportName': ''}, {'OutputKey': 'PublicSubnet1', 'OutputValue': 'subnet-af223282', 'Descri
ption': 'A reference to the public subnet in the 1st Availability Zones', 'ExportName': ''}, {'OutputKey': 'PublicSubnet2', 'OutputValue': 'subnet-fc1f4a72', 'Description': 'A reference
to the public subnet in the 2nd Availability Zones', 'ExportName': ''}, {'OutputKey': 'PrivateSubnet1', 'OutputValue': 'subnet-03d13c44', 'Description': 'A reference to the private subne
t in the 1st Availability Zones', 'ExportName': ''}, {'OutputKey': 'PrivateSubnet2', 'OutputValue': 'subnet-2444c28b', 'Description': 'A reference to the private subnet in the 2nd Availa
bility Zones', 'ExportName': ''}]", 'DBSubnetGroupName': 'sng-5783ca4b'}, 'LogicalResourceId': 'SubnetGroup', 'PhysicalResourceId': None, '_state_': {}}
```

### Expected Behavior

I expected a CSV string of subnets to be passed, something like `subnet-abcdef,subnet-12345`.

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

docker-compose.yml:

```
version: '2.1'
services:
  localstack:
    image: localstack/localstack
    ports:
      - "443:443"
      - "4510-4520:4510-4520"
      - "4566-4620:4566-4620"
      - "${PORT_WEB_UI-8080}:${PORT_WEB_UI-8080}"
    environment:
      - LOCALSTACK_API_KEY
      - DEBUG=1
      - LAMBDA_REMOTE_DOCKER=0
      - DOCKER_HOST=unix:///var/run/docker.sock
      - HOST_TMP_FOLDER=/tmp/localstack
    volumes:
      - "${TMPDIR:-/tmp/localstack}:/tmp/localstack"
      - "${HOME}/dev/ecclesia-lambdas:${HOME}/dev/ecclesia-lambdas"
      - "/var/run/docker.sock:/var/run/docker.sock"
```

`docker-compose up`

### Environment

```markdown
- OS: Ubuntu 18.04.3
- LocalStack: 


LocalStack version: 0.12.15
LocalStack build date: 2021-07-15
LocalStack build git hash: 3dc9da1a
```
```


### Anything else?

This seems related to https://github.com/localstack/localstack/issues/2079.  Full templates can be provided if needed.

**Comment by irgeek** [comment]
I think we've  run into the same bug while evaluating LocalStack Pro. Unfortunately it's a show-stopper for us as we use nested stacks extensively and outputs and parameters are how we pass values between stacks.

We've managed to create a minimal set of templates to demonstrate the problem and posted them in a gist here:

https://gist.github.com/irgeek/cd528bfce19e5d14a9d9bdd245030e4a

The templates can be deployed with commands similar to this:

```
aws cloudformation package \
  --template-file 00-vpc.yaml \
  --s3-bucket <bucket_name> \
  --output-template-file deploy.yaml
aws cloudformation deploy \
  --stack-name localstack-outputs-bug \
  --template-file deploy.yaml
```

When we run this against the real CloudFormation service the SSM Parameter (using `aws ssm get-parameter --name Subnet`) looks like this:

```json
{
    "Parameter": {
        "Name": "Subnet",
        "Type": "String",
        "Value": "subnet-01b95cc03fdac496f",
        "Version": 1,
        "LastModifiedDate": 1637122740.395,
        "ARN": "arn:aws:ssm:ap-southeast-2:952444513418:parameter/Subnet",
        "DataType": "text"
    }
}
```

When we run it against LocalStack, though, `Value` is not set correctly.

```json
{
    "Parameter": {
        "Name": "Subnet",
        "Type": "String",
        "Value": "None",
        "Version": 1,
        "LastModifiedDate": 1637123207.232,
        "ARN": "arn:aws:ssm:us-east-1:000000000000:parameter/Subnet",
        "DataType": "text"
    }
}
```

**Comment by bamajap** [comment]
We're evaluating LocalStack and this is a blocker for us too and we can't use LocalStack until it can support nested objects/outputs correctly.

**Comment by dominikschubert** [comment]
@bamajap @irgeek @redterror 

The reported bug should now be fixed and will be included in the tagged 1.4 release.
You can also pull the latest image ` docker pull localstack/localstack:latest` to try give it a try already.

Let us know if you're still seeing related issues!
